File: articles/job-satisfaction-organizational-commitment/go/satisfaction_commitment_score.go
// Job Satisfaction and Organizational Commitment
// Simple Go scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, institutional learning,
// and reproducible workflows only. Not for employment, commitment scoring,
// retention-risk scoring, productivity ranking, or individual decisions.

package main

import "fmt"

type WorkAttitudeProfile struct {
	LeadershipSupport       float64
	RewardFairness          float64
	DevelopmentalOpportunity float64
	WorkloadBalance         float64
	InstitutionalTrust      float64
	CulturalAlignment       float64
	RoleInsecurity          float64
	ExhaustionPressure      float64
}

func JobSatisfactionScore(p WorkAttitudeProfile) float64 {
	return 0.16*p.LeadershipSupport +
		0.15*p.RewardFairness +
		0.13*p.DevelopmentalOpportunity +
		0.14*p.WorkloadBalance +
		0.13*p.InstitutionalTrust +
		0.10*p.CulturalAlignment -
		0.09*p.RoleInsecurity -
		0.10*p.ExhaustionPressure
}

func main() {
	profile := WorkAttitudeProfile{84, 79, 82, 76, 83, 81, 18, 22}
	fmt.Printf("Synthetic job satisfaction score: %.2f\n", JobSatisfactionScore(profile))
	fmt.Println("Responsible-use reminder: institutional learning only.")
}

File: articles/job-satisfaction-organizational-commitment/rust/satisfaction_commitment_score.rs
// Job Satisfaction and Organizational Commitment
// Simple Rust scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, institutional learning,
// and reproducible workflows only. Not for employment, commitment scoring,
// retention-risk scoring, productivity ranking, or individual decisions.

struct WorkAttitudeProfile {
    leadership_support: f64,
    reward_fairness: f64,
    developmental_opportunity: f64,
    workload_balance: f64,
    institutional_trust: f64,
    cultural_alignment: f64,
    role_insecurity: f64,
    exhaustion_pressure: f64,
}

fn job_satisfaction_score(p: &WorkAttitudeProfile) -> f64 {
    0.16 * p.leadership_support
        + 0.15 * p.reward_fairness
        + 0.13 * p.developmental_opportunity
        + 0.14 * p.workload_balance
        + 0.13 * p.institutional_trust
        + 0.10 * p.cultural_alignment
        - 0.09 * p.role_insecurity
        - 0.10 * p.exhaustion_pressure
}

fn main() {
    let profile = WorkAttitudeProfile {
        leadership_support: 84.0,
        reward_fairness: 79.0,
        developmental_opportunity: 82.0,
        workload_balance: 76.0,
        institutional_trust: 83.0,
        cultural_alignment: 81.0,
        role_insecurity: 18.0,
        exhaustion_pressure: 22.0,
    };

    println!(
        "Synthetic job satisfaction score: {:.2}",
        job_satisfaction_score(&profile)
    );
    println!("Responsible-use reminder: institutional learning only.");
}
